Projects/Server/GameCounter.cpp

```diff
@@ -0,0 +1,96 @@
+#include "GameCounter.h"
+
+GameCounter::GameCounter(const AbstractPlayer::Ptr& player0, const AbstractPlayer::Ptr& player1, const size_t& gameSample = 1)
+	: _player0{ player0 }, _player1{ player1 }
+{
+	gameStats["p0win"];
+	gameStats["p1win"];
+	gameStats["draw"];
+	gameStats["gamesPlayed"];
+	gameStats["gameSample"] = gameSample;
+}
+
+
+void GameCounter::play() {
+
+
+	GameHandler gameHandler{};
+	size_t gameSample = gameStats["gameSample"];
+	AbstractPlayer::Ptr player0 = getPlayer0();
+	AbstractPlayer::Ptr player1 = getPlayer1();
+
+	for (size_t i{ 0 }; i < gameSample; ++i) {
+		
+		gameHandler.init(player0, player1);
+		
+
+		State::Outcome result = gameHandler.play();
+		++gameStats["gamesPlayed"];
+
+		switch (result) 
+		{
+			case State::Outcome::P0WIN:
+				++gameStats["p0win"];
+				break;
+			case State::Outcome::P1WIN:
+				++gameStats["p1win"];
+				break;
+			case State::Outcome::DRAW:
+				++gameStats["draw"];
+				break;
+			default:
+				throw std::runtime_error{ "Invalid outcome for game result." };
+		}
+		
+	}
+}
+
+std::ostream& operator<<(std::ostream& os, const GameCounter& gameCounter) 
+{	
+	os << "\n";
+	os << "Games played: " << gameCounter.getGamesPlayed() << " / " << gameCounter.getGameSample() << "\n";
+	os << "P0( " << gameCounter.getPlayer0Name() << " ) won: " << gameCounter.getP0Win() << " game";
+	os << (gameCounter.getP0Win() > 1 ? "s" : "") << "\n";
+	os << "P1( " << gameCounter.getPlayer1Name() << " ) won: " << gameCounter.getP1Win() << " game";
+	os << (gameCounter.getP1Win() > 1 ? "s" : "") << "\n";
```

Comment:
Usually no need for these irrelevant details (like adding 's' for multiple games).

Reply:
Should I remove them then? Or was this for future reference?